schemas/root.schema.json

```diff
@@ -0,0 +1,183 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "required": ["signatures", "signed"],
+  "additionalProperties": false,
+  "properties": {
+    "signatures": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "required": ["keyid", "sig"],
+        "additionalProperties": false,
+        "properties": {
+          "keyid": {
+            "type": "string",
+            "minLength": 64,
+            "maxLength": 64,
+            "pattern": "^[a-f0-9]{64}$"
+          },
+          "sig": {
+            "type": "string",
+            "minLength": 1024,
+            "maxLength": 1024,
+            "pattern": "^[a-f0-9]{1024}$"
+          }
+        }
+      }
+    },
+    "signed": {
+      "type": "object",
+      "required": ["_type", "expires", "keys", "roles", "spec_version", "version"],
+      "additionalProperties": false,
+      "properties": {
+        "_type": {
+          "enum": ["root"]
+        },
+        "consistent_snapshot": {
+          "type": "boolean"
+        },
+        "expires": {
+          "type": "string",
+          "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$"
+        },
+        "keys": {
+          "type": "object",
+          "additionalProperties": false,
+          "patternProperties": {
+            "^[0-9a-f]{64}$": {
+              "type": "object",
+              "required": ["keytype", "keyval", "scheme"],
+              "additionalProperties": true,
+              "properties": {
+                "keytype": {
+                  "enum": ["rsa"]
+                },
+                "keyval": {
+                  "type": "object",
+                  "required": ["public"],
+                  "additionalProperties": false,
+                  "properties": {
+                    "public": {
+                      "type": "string",
+                      "minLength": 1
+                    }
+                  }
+                },
+                "scheme": {
+                  "type": "string",
+                  "minLength": 1
+                }
+              }
+            }
+          }
+        },
+        "roles": {
+          "type": "object",
+          "required": ["root", "snapshot", "targets", "timestamp"],
+          "additionalProperties": false,
+          "properties": {
+            "mirror": {
+              "type": "object",
+              "required": ["keyids", "threshold"],
+              "additionalProperties": false,
+              "properties": {
+                "keyids": {
+                  "type": "array",
+                  "items": {
+                    "type": "string",
+                    "minLength": 1
```

Comment:
I think we can safely reuse the same keyid requirement as above (SHA2-256)

Reply:
Adjusted in 1e6a12972a038a7d034012430a3010377a585ef8.